torchtext/experimental/datasets/raw/translation.py

```diff
@@ -125,14 +125,26 @@ def _setup_datasets(dataset_name,
     src_eval, tgt_eval = valid_filenames
     src_test, tgt_test = test_filenames
 
-    extracted_files = []
+    extracted_files = []  # list of paths to the extracted files
     if isinstance(URLS[dataset_name], list):
         for idx, f in enumerate(URLS[dataset_name]):
-            dataset_tar = download_from_url(f, root=root, hash_value=MD5[dataset_name][idx], hash_type='md5')
+            dataset_tar = download_from_url(
+                f, root=root, hash_value=MD5[dataset_name][idx], hash_type='md5')
             extracted_files.extend(extract_archive(dataset_tar))
     elif isinstance(URLS[dataset_name], str):
-        dataset_tar = download_from_url(URLS[dataset_name], root=root, hash_value=MD5[dataset_name], hash_type='md5')
-        extracted_files.extend(extract_archive(dataset_tar))
+        dataset_tar = download_from_url(URLS[dataset_name])
```

Comment:
Do we need to check the hash here?

Reply:
Running 
`dataset_tar = download_from_url(URLS[dataset_name], root=root, hash_value=MD5[dataset_name], hash_type='md5')`

gives me this error: 

`RuntimeError: The hash of .data/2016-01.tgz does not match. Delete the file manually and retry.`

That's why I changed it to `dataset_tar = download_from_url(URLS[dataset_name])`

Why is checking the hash important and what problems could the current change introduce?

Thanks!

Reply:
Oh is this purpose of MD5 hash to stop the download if we find that the file we're downloading is not what we expected by checking the hash?

Reply:
Just made the changes!